src/screens/Notifications/UpdateEmailNotificationScreen.tsx

```diff
@@ -0,0 +1,259 @@
+  storedEmail: string;
+  updateNotificationEmail: UpdateNotificationEmailFunction;
+  updateNotificationEmailSuccess: (sessionToken: string) => UpdateNotificationEmailSuccessAction;
+}
+
+type State = {
+  email: string;
+};
+
+class UpdateEmailNotificationScreen extends PureComponent<Props, State> {
+  state = {
+    email: '',
+  };
+
+  componentDidMount() {
+    this.props.setError('');
+  }
+
+  setEmail = (email: string): void => {
+    this.props.setError('');
+    this.setState({
+      email,
+    });
+  };
+
+  goToLocalEmailConfirm = () => {
+    const { verifyNotificationEmail, navigation } = this.props;
+
+    const { email } = this.state;
+
+    verifyNotificationEmail(email, {
+      onSuccess: () =>
+        navigation.navigate(Route.LocalConfirmNotificationCode, {
+          children: (
+            <View style={styles.infoContainer}>
+              <Text style={typography.headline4}>{i18n.notifications.confirmEmail}</Text>
+              <Text style={styles.codeDescription}>{i18n.notifications.pleaseEnter}</Text>
+              <Text style={typography.headline5}>{email}</Text>
+            </View>
+          ),
+          title: i18n.notifications.notifications,
+          onSuccess: this.onUpdateSuccess,
+          email,
+        }),
+    });
+  };
+
+  onUpdateSuccess = () =>
+    this.props.createNotificationEmail(this.state.email, {
+      onSuccess: () => {
+        CreateMessage({
+          title: i18n.contactCreate.successTitle,
+          description: i18n.notifications.emailChangedSuccessMessage,
+          type: MessageType.success,
+          buttonProps: {
+            title: i18n.notifications.goToNotifications,
+            onPress: () => {
+              this.props.navigation.navigate(Route.Notifications);
+            },
+          },
+        });
+      },
+    });
+
+  authenticateNewEmail = () => {
+    const { email } = this.state;
+    const { navigation, route } = this.props;
+    const { subscribedWallets } = route.params;
+
+    return navigation.navigate(Route.ConfirmEmail, {
+      email,
+      title: i18n.notifications.confirmNewTitle,
+      description: `${i18n.notifications.verifyActionDescription} ${email}`,
+      onSuccess: this.onUpdateSuccess,
+      walletsToSubscribe: subscribedWallets,
+      onBack: this.goBackToNotificationScreen,
+    });
+  };
+
+  onConfirm = async () => {
+    const { email } = this.state;
+    const {
+      navigation,
+      route,
+      subscribableWallets,
+      setError,
+      storedEmail,
+      updateNotificationEmail,
+      updateNotificationEmailSuccess,
+    } = this.props;
+    const { subscribedWallets } = route.params;
+
+    if (!isEmail(email)) {
+      return setError(i18n.onboarding.emailValidation);
+    }
+    if (!subscribableWallets.length) {
+      return this.goToLocalEmailConfirm();
+    }
+    const hashes = await Promise.all(subscribableWallets!.map(wallet => getWalletHashedPublicKeys(wallet)));
```

Comment:
please move to sagas, I've already refactored the remaining sagas in this PR https://github.com/bitcoinvault/GoldWallet/pull/607